snappy_wrappers/wrappers/pvactools/combining/comb_rna.py

```diff
@@ -0,0 +1,530 @@
+import argparse
+import logging
+import re
+import sys
+
+import pandas as pd
+import pyranges as pr
+import vcfpy
+
+
+########################## Reading vcf files ##########################
+def create_vcf_reader(args):
+    vcf_reader = vcfpy.Reader.from_path(args.input_vcf)
+    if args.rna_vcf:
+        rna_vcf_reader = vcfpy.Reader.from_path(args.rna_vcf)
+    else:
+        rna_vcf_reader = None
+    # expression part
+    is_multi_sample = len(vcf_reader.header.samples.names) > 1
+    if is_multi_sample and args.sample_name is None:
+        vcf_reader.close()
+        raise Exception(
+            "ERROR: VCF {} contains more than one sample. Please use the -s option to specify which sample to annotate.".format(
+                args.input_vcf
+            )
+        )
+    elif is_multi_sample and args.sample_name not in vcf_reader.header.samples.names:
+        vcf_reader.close()
+        raise Exception(
+            "ERROR: VCF {} does not contain a sample column for sample {}.".format(
+                args.input_vcf, args.sample_name
+            )
+        )
+    if "CSQ" not in vcf_reader.header.info_ids():
+        vcf_reader.close()
+        raise Exception(
+            "ERROR: VCF {} is not VEP-annotated. Please annotate the VCF with VEP before running this tool.".format(
+                args.input_vcf
+            )
+        )
+    if args.mode == "gene" and "GX" in vcf_reader.header.format_ids():
+        vcf_reader.close()
+        raise Exception(
+            "ERROR: VCF {} is already gene expression annotated. GX format header already exists.".format(
+                args.input_vcf
+            )
```

Comment:
The header may also have the GX entry without any record actually having that field set!